Mathlib/Combinatorics/SimpleGraph/Diam.lean

```diff
@@ -131,22 +198,12 @@ lemma ediam_bot [Nontrivial α] : (⊥ : SimpleGraph α).ediam = ⊤ :=
 
 @[simp]
 lemma ediam_top [Nontrivial α] : (⊤ : SimpleGraph α).ediam = 1 := by
-  apply le_antisymm ?_ <| Order.one_le_iff_pos.mpr <| pos_iff_ne_zero.mpr ediam_ne_zero
-  apply ediam_def ▸ iSup_le_iff.mpr
-  intro p
-  by_cases h : (⊤ : SimpleGraph α).Adj p.1 p.2
-  · apply le_of_eq <| edist_eq_one_iff_adj.mpr h
-  · simp_all
+  simp [ediam]
 
 @[simp]
-lemma ediam_eq_one [Nontrivial α] : G.ediam = 1 ↔ G = ⊤ := by
-  refine ⟨fun h₁ ↦ ?_, fun h ↦ h ▸ ediam_top⟩
-  ext u v
-  refine ⟨fun h ↦ h.ne, fun h₂ ↦ ?_⟩
-  apply G.edist_pos_of_ne at h₂
-  apply le_of_eq at h₁
-  rw [ediam_def, iSup_le_iff] at h₁
-  exact edist_eq_one_iff_adj.mp <| le_antisymm (h₁ (u, v)) <| Order.one_le_iff_pos.mpr h₂
+lemma ediam_eq_one [Nontrivial α] : G.ediam = 1 ↔ G = ⊤ :=
+  ⟨fun h ↦ eq_top_iff_forall_eccent_eq_one.mpr fun u ↦ le_antisymm (h.symm ▸ eccent_le_ediam u)
+  <| Order.one_le_iff_pos.mpr <| pos_iff_ne_zero.mpr <| eccent_ne_zero u, fun h ↦ h ▸ ediam_top⟩
```

Comment:
Tactic mode for this one too, please! Nice one-line term mode proofs are nice, but this one is too long to count as one of those!